### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by MayerLansky**
[DEMO LINK](https://MayerLansky.github.io/layout_catalog/)
[TEST REPORT LINK](https://MayerLansky.github.io/layout_catalog/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -5,8 +5,323 @@
...
+    <header class="header">
+      <a href="#home">
+        <img 
+          src="./images/logo.png" 
+          class="header__logo" 
+          alt="moyo"
+        >
+      </a>
+      <nav class="nav">
+        <ul class="nav_list">
```
Comment (vlad-demchuk) [inline src/index.html:23]:
```suggestion
        <ul class="nav__list">
```
follow bem naming convention
https://en.bem.info/methodology/naming-convention/#naming-rules

File: src/index.html [hunk 2]
```diff
@@ -5,8 +5,323 @@
...
+          <li class="links">
```
Comment (vlad-demchuk) [inline src/index.html:24]: no need to create a separate bem block for links here

File: src/index.html [hunk 3]
```diff
@@ -5,8 +5,323 @@
...
+      </nav>   
+    </header>
+    
+    <main class="catalog">
+      <div class="card" data-qa="card">
+        <img 
+        src="./images/imac.jpeg" 
+        class="card__img" 
+        alt="APPLE A1419 iMac 27"
+      >
```
Comment (vlad-demchuk) [inline src/index.html:65]:
```suggestion
      <img 
        src="./images/imac.jpeg" 
        class="card__img" 
        alt="APPLE A1419 iMac 27"
      >
```

File: src/index.html [hunk 4]
```diff
@@ -5,8 +5,323 @@
...
+          <p class="card__price-value">$2,199</p>
+        </div>
+
+        <a 
+        href="#buy" 
+        class="card__buy-link" 
+        data-qa="hover"
+        >
+        buy
+        </a>
```
Comment (vlad-demchuk) [inline src/index.html:123]:
```suggestion
        <a 
          href="#buy" 
          class="card__buy-link" 
          data-qa="hover"
        >
        buy
        </a>
```

Comment (vlad-demchuk) [reply src/index.html:123]: check your code style
fix all such cases

File: src/styles/main.scss [hunk 5]
```diff
@@ -1,3 +1,243 @@
...
+$card-margin-right: 48px;
+$padding-horizont: 41px;
+$padding-vertical: 51px;
+$card-margin-bottom: 46px;
+$width: ($width-card * 4) + ($card-margin-right * 3) + (2 * $padding-horizont);
+$head-padding: 0 50px;
+$link-rightmarg: 20px;
+$link-padding: 23px 0;
+$font-ten: 10px;
+$font-twelve: 12px;
```
Comment (vlad-demchuk) [inline src/styles/main.scss:15]: split your styles code to separate files:

bem blocks, scss features like variables, mixins and so on, reset file

File: src/styles/main.scss [hunk 6]
```diff
@@ -1,3 +1,243 @@
...
+
+html {
+  font-family: Roboto, sans-serif;
+}
+
 body {
   margin: 0;
+  padding: 0;
+  width: 100%;//?????????
+  min-width: $width;//???????
```
Comment (vlad-demchuk) [inline src/styles/main.scss:29]: remove comments

File: src/styles/main.scss [hunk 7]
```diff
@@ -1,3 +1,243 @@
...
+.header {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  background: #fff;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.05);
+  width: 100%;
+  // height: 60px;
+  // min-width: 1024px; //?????
```
Comment (vlad-demchuk) [inline src/styles/main.scss:42]: remove all comments

File: src/styles/main.scss [hunk 8]
```diff
@@ -1,3 +1,243 @@
...
+    text-decoration: none;
+    text-transform: uppercase;
+    color: #060b35;
+    padding: $link-padding;
+    font-weight: 500;
+    font-size: $font-twelve;
+    line-height: 14px; //?????????????
+  }
+
+  &__link_a:active::after {
```
Comment (vlad-demchuk) [inline src/styles/main.scss:78]: you already have .is-active class
why do you duplicate these styles?

File: src/styles/main.scss [hunk 9]
```diff
@@ -1,3 +1,243 @@
...
+
+.catalog {
+  display: flex;
+  justify-content: center;
+  flex-wrap: wrap;
+  padding: $padding-vertical $padding-horizont;
+  // padding: 50px 0;
+  // width: 944px;
+  min-width: $width;
+  margin: 0 auto;
```
Comment (vlad-demchuk) [inline src/styles/main.scss:118]: image

File: src/styles/main.scss [hunk 10]
```diff
@@ -1,3 +1,243 @@
...
+
+.card {
+  width: $width-card;
+  border: 1px solid #{$color-bck};
+  border-radius: 5px;
+  display: flex;
+  flex-direction: column;
+  padding: 16px;
+  margin-right: $card-margin-right;
+  margin-bottom: $card-margin-bottom;//????????????? the second line 73
```
Comment (vlad-demchuk) [inline src/styles/main.scss:131]: image

**Review by vlad-demchuk — CHANGES_REQUESTED** [review]
implementation doesn't match the mock-up, most of tests failed
try to fix them

if you have any questions, just feel free to ask in fe_chat

**Comment by MayerLansky** [comment]
I know that. I deployed it for checking by Misha Djemula. I can’t figure
out how it must be working.

**Review by DarkMistyRoom — COMMENTED** [review]
Try to fix all tests:
- hardcode the sizes of the log to avoid the bottom indent

image

- check the indents between the elements inside the card. And be consistent with your margins - add only top or bottom, don't add both.

image

What exactly do you want me to check? If you deploy for checking while the task isn't finished yet, there is no need to make PR - just deploy. Write in the chat if you need help

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]
Please write in the chat, what kind of help do you need

File: src/index.html [hunk 11]
```diff
@@ -5,8 +5,302 @@
...
+            <p class="card__price-label">Price:</p>
+            <p class="card__price-value">$2,199</p>
+          </div>
+          <a
+          href="#buy"
+          class="card__buy-link"
+          data-qa="hover"
+          >
+          buy
+          </a>
```
Comment (anstsot) [inline src/index.html:149]: fix the code formatting
```suggestion
          <a
            href="#buy"
            class="card__buy-link"
            data-qa="hover"
          >
            buy
          </a>
```

Comment (anstsot) [reply src/index.html:149]: please check the whole html file

File: src/styles/main.scss
Comment (anstsot) [inline src/styles/main.scss:1]: please create additional files
image

**Review by anstsot — CHANGES_REQUESTED** [review]
Try to fix all tests. If you have any questions, feel free to ask them in the chat

File: src/index.html [hunk 12]
```diff
@@ -5,8 +5,302 @@
...
+          <p class="card__type">Product code: 195434</p>
+          <div class="card__review-container">
+            <div class="stars stars--4">
+              <div class="stars__star"></div>
+              <div class="stars__star"></div>
+              <div class="stars__star"></div>
+              <div class="stars__star"></div>
+              <div class="stars__star"></div>
+            </div>
+            <div class="card__review">Reviews: 5</div>
```
Comment (volodymyr-soltys97) [inline src/index.html:77]: Use a semantic tag everywhere for text
```suggestion
            <p class="card__review">Reviews: 5</p>
```

File: src/index.html (hunk 11 above)
Comment (volodymyr-soltys97) [inline src/index.html:149]: Need to fix it everywhere
```suggestion
          <a
            href="#buy"
            class="card__buy-link"
            data-qa="hover"
          >
            buy
          </a>
```

File: src/styles/main.scss [hunk 13]
```diff
@@ -1,3 +1,255 @@
...
+}
+  
+.is-active::after {
+  content: "";
+  position: absolute;
+  height: 4px;
+  left: 0;
+  bottom: 0;
+  width: 100%;
+  background: #00acdc;
```
Comment (volodymyr-soltys97) [inline src/styles/main.scss:88]: Create a variable for this color and use it everywhere

File: src/styles/main.scss [hunk 14]
```diff
@@ -1,3 +1,255 @@
...
+}
+
+.main__container {
+  display: flex;
+  flex-wrap: wrap;
+  margin-bottom: 73px;
+  // width: $width;
+  
+  // padding: 51px 41px 73px;
+
```
Comment (volodymyr-soltys97) [inline src/styles/main.scss:110]: Remove comments everywhere
```suggestion
```

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code

**Comment by MayerLansky** [comment]
Please stop doing this! Don’t you see that I’ve deployed this task 5 times?
I know how to deal with tags. Anastasia told me to deploy it to take look
on my scss folder: width, min-width, media etc.

**Pull request closed without merging**